ztools/http/request.go

```diff
@@ -449,7 +450,11 @@ func NewRequestWithHost(method, urlStr, host string, body io.Reader) (*Request,
 	}
 
 	if host == "" {
-		host = u.Host
+		if hostWithoutPort, _, err := net.SplitHostPort(u.Host); err != nil {
```

Comment:
It just returns the host if there is no port number

On Sep 10, 2016, 9:55 PM -0500, Zakir Durumeric <notifications@github.com>, wrote:
>
> In ztools/http/request.go (https://github.com/zmap/zgrab/pull/182#discussion_r78287052):
>
> > @@ -449,7 +450,11 @@ func NewRequestWithHost(method, urlStr, host string, body io.Reader) (*Request, > } > > if host == "" { > - host = u.Host > + if hostWithoutPort, _, err := net.SplitHostPort(u.Host); err != nil {
>
> So what happens here if there isn't a port number? Can that ever happen?
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub (https://github.com/zmap/zgrab/pull/182/files/32a65cf894bca465b70f739f1d8dc0df0b434d20#r78287052), or mute the thread (https://github.com/notifications/unsubscribe-auth/AC24rHNF7FWTB4Ly9nRQuN00MWESVNF1ks5qo22egaJpZM4J5z5v).
>
>
>